src/entities/author.rb

```diff
@@ -0,0 +1,19 @@
+require_relative 'entity'
+
+class Author < Entity
+  attr_reader :name, :biography
+
+  def initialize(name, biography = nil)
+    @name = name
+    @biography = biography
+    super()
+    validate
+  end
+
+  private
+
+  def validate
+    check_type(name, String)
+    check_not_empty(name)
```

Comment:
Validation works strangely if you add the wrong type.
For example, if `integer` has been passed, we got:
```
undefined method `empty?' for 111:Integer (NoMethodError)
```

Reply:
fixed